server/bin/pbench-verify-indexed.py

```diff
@@ -0,0 +1,192 @@
+    except pbench.BadConfig as e:
+        print(f"{_NAME_}: {e}", file=sys.stderr)
+        return 1
+
+    try:
+        archive_p = Path(config.ARCHIVE).resolve(strict=True)
+    except FileNotFoundError:
+        print(
+            f"The configured ARCHIVE directory, {config.ARCHIVE}," " does not exist",
+            file=sys.stderr,
+        )
+        return 1
+
+    if not archive_p.is_dir():
+        print(
+            f"The configured ARCHIVE directory, {config.ARCHIVE},"
+            " is not a valid directory",
+            file=sys.stderr,
+        )
+        return 1
+
+    es_client = Elasticsearch(
+        pbench.indexer._get_es_hosts(config, logging.getLogger("pbench-verify-indexed"))
+    )
+    query = {"query": {"match_all": {}}}
+
+    index_prefix = config.get("Indexing", "index_prefix")
+    if not index_prefix:
+        print(
+            "Missing 'index_prefix' value in 'Indexing' section of configuration",
+            file=sys.stderr,
+        )
+        return 1
+
+    # Start the overall timer.
+    start = pbench._time()
+    start_s = pbench.tstos(start)
+    print(f"[{start_s}] Verifying indexed state of tar balls", flush=True)
+
+    # Fetch a thousand records at a time.
+    scanner = helpers.scan(
+        es_client,
+        index=f"{index_prefix}.v4.run.*",
+        doc_type="pbench-run",
+        query=query,
+        expand_wildcards="open",
+        fields=["@metadata.file-name"],
+        size=BATCH_SIZE,
+    )
+    indexed = defaultdict(int)
+    for hit in scanner:
+        tb_file_p = Path(hit["fields"]["@metadata.file-name"][0])
+        tb_file = f"{tb_file_p.parent.name}/{tb_file_p.name}"
+        indexed[tb_file] += 1
+    indexed_cnt = len(indexed)
+
+    # Snap shot of query time.
+    q_end = pbench._time()
+
+    on_disk = defaultdict(int)
+    on_disk_cnt = 0
+    on_disk_indexed_cnt = 0
+    not_indexed = []
+    for ctrl, tb_name in gen_tb_list(archive_p):
+        tb_file = f"{ctrl}/{tb_name}"
+        on_disk[tb_file] += 1
+        on_disk_cnt += 1
+        if tb_file in indexed:
+            on_disk_indexed_cnt += 1
+        else:
+            not_indexed.append(tb_file)
+
+    # Snap shot of disk scan time.
+    d_end = pbench._time()
+
+    not_on_disk = [f for f in indexed.keys() if f not in on_disk]
+    not_on_disk_cnt = len(not_on_disk)
+
+    # The heavy lifting is done.
+    end = pbench._time()
+    end_s = pbench.tstos(end)
+
+    print(f"[{end_s}] Generating report ...")
+    print(f"On Disk: {on_disk_cnt:n}, indexed {on_disk_indexed_cnt:n}")
+    print(f"Indexed: {indexed_cnt:n}, not on disk {not_on_disk_cnt:n}")
+    print(
+        f"Run-time: {(end - start):.2f}s, query time {(q_end - start):.2f}s,"
+        f" disk time {(d_end - q_end):.2f}s, final pass {(end - d_end):.2f}s"
+    )
+
+    if not_on_disk:
+        print(f"Not on disk... ({not_on_disk_cnt:n})")
+        for tb_file in sorted(not_on_disk):
+            print(f"\t{tb_file}")
```

Comment:
Consider replacing this with
```
        print("\t".join(sorted(not_on_disk)))
```
Ditto for lines 166-167.

Reply:
First entry wouldn't have a leading tab.  Since buffered I/O is used by default anyway it doesn't really matter.

Reply:
> First entry wouldn't have a leading tab.

Oh, sorry:
```
        print("\t" + "\t".join(sorted(not_on_disk)))
```
😁

> Since buffered I/O is used by default anyway it doesn't really matter.

I have no idea whether it matters or not at execution time -- we would have to do some sort of trial.  But, in terms of the source code, there is a difference...although, whether it's an improvement or not is perhaps in the eye of the beholder.